pkg/service/repair/generator.go

```diff
@@ -125,19 +125,20 @@ func (g *generator) Run(ctx context.Context) (err error) {
 	g.logger.Info(ctx, "Start generator")
 	var genErr error
 
-	defer func() {
-		// Always leave tablet migration enabled after repair
-		tabletBalancingErr := g.ringDescriber.ControlTabletLoadBalancing(context.Background(), true)
-		err = stdErrors.Join(err, errors.Wrap(tabletBalancingErr, "control post repair tablet load balancing"))
-	}()
+	if !g.plan.apiSupport.tabletRepair {
+		defer func() {
+			// Always leave tablet migration enabled after repair
+			tabletBalancingErr := g.ringDescriber.ControlTabletLoadBalancing(context.Background(), true)
+			err = stdErrors.Join(err, errors.Wrap(tabletBalancingErr, "control post repair tablet load balancing"))
+		}()
+	}
 
 	for _, ksp := range g.plan.Keyspaces {
-		if !g.plan.apiSupport.tabletRepairNoHostFiltering {
+		if !g.plan.apiSupport.tabletRepair {
 			// Disable tablet migration when repairing tablet table.
 			// Without that it could be possible that some tablet "escapes" being
 			// repaired by migrating from not yet repaired token range to already repaired one.
-			// We don't need to do it when tablet repair API is available
-			// (even if it does not allow for host filtering).
+			// We don't need to do it when tablet repair API is available.
 			if err := g.ringDescriber.ControlTabletLoadBalancing(ctx, g.ringDescriber.IsTabletKeyspace(ksp.Keyspace)); err != nil {
 				return errors.Wrapf(err, "control tablet load balancing")
 			}
```

Comment:
My understanding is:
- per every ks, if Scylla API doesn't support tablet's API then 
   - KS is tablet based => keep the load-balancing enabled
   - KS is vnodes based => disable load-balancing
  
It it opposite to the comment:

> 			// Disable tablet migration when repairing tablet table.
			// Without that it could be possible that some tablet "escapes" being
			// repaired by migrating from not yet repaired token range to already repaired one.
			// We don't need to do it when tablet repair API is available.

Expected ? Or my understanding is wrong ?

Reply:
It looks like there is missing `!` when checking if keyspace is tablet replicated, and unfortunately it is not introduced by this PR, but it was like this from the very beginning....
The good thing is that tablet's tombstone gc mode repair protects us from things like data resurrection, but this is still a big issue that wasn't caught by any test.
I will fix it in this PR and add some tests for it as well.